main.tf

```diff
@@ -355,3 +357,80 @@ resource "lacework_integration_aws_ct" "default" {
 
   depends_on = [time_sleep.wait_time]
 }
+
+data "aws_iam_policy_document" "s3_notification_relay_assume" {
+  count = (var.use_existing_cloudtrail && var.use_s3_notification_relay) ? 1 : 0
+
+  statement {
+    effect = "Allow"
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+    actions = ["sts:AssumeRole"]
+  }
+}
+
+data "aws_iam_policy_document" "s3_notification_relay_sns" {
+  count = (var.use_existing_cloudtrail && var.use_s3_notification_relay) ? 1 : 0
+
+  statement {
+    effect    = "Allow"
+    actions   = ["sns:Publish"]
+    resources = [local.sns_topic_arn]
+  }
+}
+
+resource "aws_iam_role" "s3_notification_relay" {
+  count = (var.use_existing_cloudtrail && var.use_s3_notification_relay) ? 1 : 0
+
+  name        = local.s3_notification_role_name
+  description = "Allows Lambda function for call AWS services"
+
+  assume_role_policy = data.aws_iam_policy_document.s3_notification_relay_assume[count.index].json
+  inline_policy {
+    name   = "${local.s3_notification_role_name}-policy"
+    policy = data.aws_iam_policy_document.s3_notification_relay_sns[count.index].json
+  }
+  managed_policy_arns = ["arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"]
+}
+
+resource "aws_lambda_permission" "s3_notification_relay_notify" {
+  count = (var.use_existing_cloudtrail && var.use_s3_notification_relay) ? 1 : 0
+
+  statement_id  = "AllowExecutionFromS3"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.s3_notification_relay[count.index].arn
+  principal     = "s3.amazonaws.com"
+  source_arn    = local.bucket_arn
+}
+
+resource "aws_s3_bucket_notification" "s3_notification_relay" {
+  count = (var.use_existing_cloudtrail && var.use_s3_notification_relay) ? 1 : 0
+
+  bucket = local.bucket_name
+
+  lambda_function {
+    lambda_function_arn = aws_lambda_function.s3_notification_relay[count.index].arn
+    events              = ["s3:ObjectCreated:*"]
+    filter_prefix       = "AWSLogs/"
+  }
+}
+
+resource "aws_lambda_function" "s3_notification_relay" {
+  count = (var.use_existing_cloudtrail && var.use_s3_notification_relay) ? 1 : 0
+
+  function_name = local.s3_notification_lambda_name
+  role          = aws_iam_role.s3_notification_relay[count.index].arn
+  handler       = "main.handler"
+  runtime       = "python3.9"
+
+  environment {
+    variables = {
+      "SNS_TOPIC_ARN" = local.sns_topic_arn
+    }
+  }
+
+  filename         = "${path.module}/functions/packages/s3-notification-relay/main.zip"
+  source_code_hash = filebase64sha256("${path.module}/functions/packages/s3-notification-relay/main.zip")
+}
```

Comment:
I would suggest moving this to a different plan file since it is very specific to s3 notifications/lambda, it might confuse folks seeing all of these up front (silly but true)